data/json/monsters/insect_spider.json

```diff
@@ -401,62 +401,179 @@
-    "extend": { "flags": [ "DESTROYS", "PUSH_MON", "PUSH_VEH" ] }
+    "hp": 200,
+    "volume": "300 L",
+    "weight": "400 kg",
+    "speed": 60,
+    "aggression": 0,
+    "morale": 10,
+    "melee_skill": 3,
+    "melee_dice_sides": 5,
+    "special_attacks": [
+      [ "GRAB", 8 ],
+      {
+        "type": "bite",
+        "move_cost": 120,
+        "cooldown": 5,
+        "damage_max_instance": [ { "damage_type": "stab", "amount": 10, "armor_penetration": 12 } ],
+        "min_mul": 1,
+        "max_mul": 2.1,
+        "no_infection_chance": 25,
+        "effects": [ { "id": "venom_dmg", "duration": 500, "affect_hit_bp": true } ]
+      },
+      {
+        "type": "spell",
+        "spell_data": { "id": "centipede_mom_summon" },
+        "monster_message": "A small centipede is shaken off in the struggle!",
+        "cooldown": 20
+      }
+    ],
+    "dodge": 0,
+    "proportional": { "vision_day": 0.5, "vision_night": 0.5 },
+    "anger_triggers": [ "HURT", "PLAYER_CLOSE" ],
+    "harvest": "arachnid_centipede_mom",
+    "reproduction": { "baby_monster": "mon_centipede_small", "baby_count": 2, "baby_timer": 10 },
+    "baby_flags": [ "SPRING", "SUMMER", "AUTUMN" ],
+    "upgrades": { "age_grow": 15, "into_group": "GROUP_CENTIPEDE_MOM" },
+    "delete": { "flags": [ "KEENNOSE", "GOODHEARING" ] }
+  },
+  {
+    "id": "mon_centipede_mega",
+    "type": "MONSTER",
+    "name": { "str": "armored centipede" },
+    "description": "A mutated centipede covered in interlocking chitinous plates of metallic grey, reminding you more of a runaway train car than any living animal.  Its venom claws end in jagged points, seemingly specialised in hunting prey at least as well-armored as itself.  You really don't want to try if you'd fare better.",
+    "symbol": "A",
+    "color": "dark_gray",
+    "default_faction": "centipede",
+    "bodytype": "insect",
+    "material": [ "iflesh" ],
+    "species": [ "CENTIPEDE" ],
+    "volume": "950 L",
+    "weight": "2500 kg",
+    "hp": 850,
+    "speed": 110,
+    "attack_cost": 110,
+    "aggression": 100,
+    "regen_morale": true,
+    "melee_skill": 6,
+    "melee_dice": 3,
+    "melee_dice_sides": 6,
+    "melee_damage": [ { "damage_type": "stab", "amount": 15, "armor_penetration": 25 } ],
+    "special_attacks": [
+      [ "GRAB_DRAG", 12 ],
+      {
+        "type": "bite",
+        "cooldown": 5,
+        "move_cost": 100,
+        "damage_max_instance": [ { "damage_type": "stab", "amount": 20, "armor_penetration": 110 } ],
+        "min_mul": 0.8,
+        "max_mul": 1.1,
+        "no_infection_chance": 30,
+        "effects": [ { "id": "venom_dmg", "duration": 750, "affect_hit_bp": true } ]
+      }
+    ],
+    "//": "3 hits to max intensity, 4 to max duration",
+    "armor_bash": 25,
+    "armor_bullet": 45,
+    "armor_cut": 35,
+    "armor_stab": 15,
+    "vision_day": 12,
+    "vision_night": 7,
+    "anger_triggers": [ "STALK", "PLAYER_WEAK", "HURT" ],
```

Comment:
Not PLAYER_CLOSE (which the others have)?

Reply:
Monster morale is a *mess*, and this is the result of a lot of hair-tearing tweaking: I want them to hold at the edge of a fire, *not* run away and *not* attacking until the fire is gone out.
If I give them PLAYER_CLOSE, then they stop get angry enough to storm past fire.
If I give them a higher morale, they just ignore the fire.
If I don't give them HURT, you could shoot them to death from your campfire.
If I don't give them morale_regen they get scared by the fire and run from you forever (well, until you reload them).

Reply:
Ouch! (Long-term, looking at [this and more-recent material](http://roguebasin.roguelikedevelopment.org/index.php?title=Category:AI) may be helpful.)